src/colmap/scene/database.h

```diff
@@ -177,11 +186,15 @@ class Database {
   // `image_id1` and `image_id2` does not matter.
   void WriteKeypoints(image_t image_id,
                       const FeatureKeypoints& keypoints) const;
+  void WriteKeypoints(image_t image_id, const FeatureKeypointsBlob& blob) const;
   void WriteDescriptors(image_t image_id,
                         const FeatureDescriptors& descriptors) const;
   void WriteMatches(image_t image_id1,
                     image_t image_id2,
                     const FeatureMatches& matches) const;
+  void WriteMatches(image_t image_id1,
+                    image_t image_id2,
+                    FeatureMatchesBlob blob) const;
```

Comment:
```suggestion
                    const FeatureMatchesBlob& blob) const;
```

Reply:
`SwapFeatureMatchesBlob` a few lines later modifies it in-place, so better copy in the argument.